src/qililab/result/qprogram/qprogram_results.py

```diff
@@ -35,3 +38,53 @@ def append_result(self, bus: str, result: MeasurementResult):
         if bus not in self.results:
             self.results[bus] = []
         self.results[bus].append(result)
+
+
+def probabilities(qprogram_results: QProgramResults, qubit_mapping: list[str] | None = None) -> dict[str, float]:
```

Comment:
I had the impression from our last meeting that this functionality would be outside of the `QProgramResults`, since it is a special case anyway.

Ramiro called it with a name, but can't remember, let's call it XYZ. The main component of this class is this matrix he described.

Then it's better to have a class called XYZ that contains the representation. The main advantage is that this class can be used independently of QProgram: the user might create similar results with other methods, either circuit or manually.

The class will also contain a class method `from(qprogram_results: QProgramResults, qubit_mapping: list[str] | None = None`) to do what this function implements.

If you think the XYZ class is the `Counts` class that we already have, implement the `from()` method there.

Reply:
Oh yeah, It is not part of the class, (it is not indented). I just left it there because I didn't know if I should move it to a new file, but yeah, I can definitely move it into a new class or maybe add it to the `Counts` module. 🙌

Reply:
In the end I moved it out to a static method of the `Counts` class